modules/custom-status/custom-status.php

```diff
@@ -582,25 +627,28 @@ public function add_custom_status( array $args ): WP_Term|WP_Error {
 	 * @param array $args Any arguments to be updated
 	 * @return object $updated_status Newly updated status object
 	 */
-	public function update_custom_status( int $status_id, array $args = [] ): WP_Term|WP_Error {
-		$old_status = $this->get_custom_status_by( 'id', $status_id );
+	public static function update_custom_status( int $status_id, array $args = [] ): WP_Term|WP_Error {
+		$old_status = self::get_custom_status_by( 'id', $status_id );
 		if ( is_wp_error( $old_status ) ) {
 			return $old_status;
 		} else if ( ! $old_status ) {
 			return new WP_Error( 'invalid', __( "Custom status doesn't exist.", 'vip-workflow' ) );
 		}
 
 		// Reset our internal object cache
-		$this->custom_statuses_cache = [];
+		self::$custom_statuses_cache = [];
+
+		// Don't allow changing the name or slug of the Draft or pending status as they are core statuses
+		$banned_statuses = [ 'draft', 'pending' ];
 
 		// Prevent user from changing draft name or slug
-		if ( 'draft' === $old_status->slug
+		if ( in_array( $old_status->slug, $banned_statuses )
 		&& (
 			( isset( $args['name'] ) && $args['name'] !== $old_status->name )
 			||
 			( isset( $args['slug'] ) && $args['slug'] !== $old_status->slug )
 		) ) {
-			return new WP_Error( 'invalid', __( 'Changing the name and slug of "Draft" is not allowed', 'vip-workflow' ) );
+			return new WP_Error( 'restricted', __( 'Changing the name and slug of a restricted status ', 'vip-workflow' ) . '(' . $old_status->name . ') is not allowed.' );
```

Comment:
Changed this message to be a lot more clearer

Reply:
I like this improvement! We can utilize i18n functions here to avoid string concats and i18n mixing:

```suggestion
			// translators: %s: Post type, like "Draft"
			return new WP_Error( 'restricted', sprintf( __( 'Changing the name and slug of a restricted status (%s) is not allowed.', 'vip-workflow' ), $old_status->name ) );
```

Reply:
Also the same thing below in `delete_custom_status()`!

Reply:
Good point, got that change in.